test/units/testsuite-13.sh

```diff
@@ -60,6 +60,12 @@ function check_notification_socket {
     systemd-nspawn --register=no -D /testsuite-13.nc-container -U /bin/sh -x -c "$_cmd"
 }
 
+function check_os_release {
+    local _cmd='busybox env | busybox grep -F -f /tmp/host-os-release'
```

Comment:
Careful: this is only asserting that at least one line from the host `os-release` appears in the guest's environment!

It also doesn't take quoting into account. The combination of those two factors is that it's relying on the host `os-release` containing at least one line that isn't quoted. In practice `ID` often isn't, but I don't think the specification requires that - `ID="debian"` is just as valid as `ID=debian`.

Ideally, the assertion should be something like this: if you `. /tmp/host-os-release` or `eval "$(grep "^NAME=" /tmp/host-os-release)"` inside the container, then for each of the known fields (or at least a representative selection of them), if the field (e.g. `PRETTY_NAME`) is set, then the prefixed version (e.g. `container_host_PRETTY_NAME`) is also set, with the same value. Perhaps something like this:

```
sh -c '
assert () {
  if ! test "$@"; then
    echo "Assertion failed: test $*"
    exit 1
  fi
}
. /tmp/host-os-release
if [ -n "${NAME:+set}" ]; then
  assert "${NAME}" = "${container_host_NAME}"
fi
... etc. ...
'
```

(including at least one field that commonly contains spaces, like `NAME` or `PRETTY_NAME`).

Reply:
Thanks for the heads-up - fixed to iterate on each line individually, and to eval away the quotes.

Given the image that runs these tests is built from whatever is on the machine of a developer/CI system on-the-fly, I'd rather not hard-code any specific key to check for, to avoid false positives in case of strange and custom developer environments.

Reply:
> Given the image that runs these tests is built from whatever is on the machine of a developer/CI system on-the-fly, I'd rather not hard-code any specific key to check for

Sure, that's why I said: *if* it's set.

Reply:
Ok, no problem - the current iteration ensures that every line is present with the prefix, and also specifically checks for NAME if it is set as you suggested. Any other key you would like to add to the list?